src/action/finder/glob_to_regex.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2020 Google LLC
+//
+// Use of this source code is governed by an MIT-style license that can be found
+// in the LICENSE file or at https://opensource.org/licenses/MIT.
+
+use crate::session::RegexParseError;
+use regex::Regex;
+
+/// Converts glob expression into a Rust Regex.
+/// E.g. "asd*[!123]??" will be converted into "asd.*[^123]..".
+/// This implementation is a Rust port of the cpython code:
+/// https://github.com/python/cpython/blob/2.7/Lib/fnmatch.py
+fn glob_to_regex(pat: &str) -> Result<Regex, RegexParseError> {
```

Comment:
GRR uses `str` to represent paths and this is a very big mistake that was made, in RRG we should make things *right* from the beginning. We should really think hard how to tackle this problem.

Reply:
I believe I understand the problem (at least partially). I designed the code to make a path transition to other data types as trivial as possible.

In the current version of the proto definition we use the input FileFinderArgs::paths contains `std::string::String` already` and I believe this is the first problem we should address.

I'm more than happy to meet to consider options on using different types for representing paths.